Implement open-link command and refactor add-link and delete-link commands (#104)
AY2223S1-CS2103T-T10-1/tp: docs/UserGuide.md

* `la/`: To be followed by the link alias which is associated with the link

Format: `add-link m/MODULE_CODE l/LINK_URL la/LINK_ALIAS`
* You cannot add a link to a non-existent module code.
* You cannot add a link using a link alias that already exists in the module represented by the module code.
* You cannot add a link using a link URL that already exists in the module represented by the module code.

Example:
```
add-link m/CS2040 l/visualgo.net/en la/visualgo
```
In the above example, we are adding the link with the URL `visualgo.net/en`
to the module with module code `CS2040`, represented by the link alias `visualgo`.

```
add-link m/CS2040 l/https://www.nusmods.com l/open.kattis.com la/nus mods la/kattis
```
In the above example, we are adding the links with the URL `https://www.nusmods.com` and `open.kattis.com`
to the module with module code `CS2040`, represented by the link alias `nus mods` and `kattis` respectively.

#### 2.3.2. Delete link
You may delete link/s from a specific module using the `delete-link` command. 
Links will be deleted by means of their corresponding alias.

Multiple links can be deleted at once using their corresponding alias. 

> Shall a link alias from a chained command be detected as invalid,
> none of the links in the command will be deleted.

Reviewer: ```suggestion
if there exist a link alias that is detected as invalid in a chained command,
none of the links in the command will be deleted.
```

weird usage of shall

This command will require two flags:
* `m/`: To be followed by the module code of the module which is associated with the link
* `la/`: To be followed by the link alias which is associated with the link

Format: `delete-link m/MODULE_CODE la/LINK_ALIAS`
* You cannot delete a link from a non-existent module code.
* You cannot delete a link using a non-existent link alias from an existing module.

Example:
```
delete-link m/CS2040 la/visualgo
```
In the above example, we are deleting the link with the URL `visualgo.net/en`
from the module with module code `CS2040`, using its corresponding link alias `visualgo`.

```
delete-link m/CS2040 la/nus mods la/kattis
```
In the above example, we are deleting the links with the URL `https://www.nusmods.com` and `open.kattis.com`
from the module with module code `CS2040`, using their corresponding link alias `nus mods` and `kattis` respectively.

#### 2.3.3. Open link
You may open link/s from a specific module to your default browser using the `open-link` command.
Links will be opened by means of their corresponding alias.

An alternative way to open links is by means of clicking on their aliases on the application window.

> Permissions from your operating system may be required for some users to use this feature.

Reviewer: Perhaps can be shown as an alert to highlight this issue to user

Multiple links can be opened at once using its corresponding alias according to their order of input
(left to right). Shall a link alias from a chained command be detected as invalid,

> the links to its left will be opened while the links to its right will not be opened.

Reviewer: Same as aforementioned issue regarding weird usage of shall
